### ISSUE TITLE
filePath or moduleId (or both)?

### ISSUE THREAD

**Issue by jgwhite** [ember-template-lint/ember-template-lint#1346]
The README suggests using the linter like this:

```js
var results = linter.verify({ source: template, moduleId: 'template.hbs' });
```

And indeed, [this is how ember-language-server uses it](https://github.com/ember-tooling/ember-language-server/blob/db75e45744e8da8023564385a68ca0f2dc0468d9/src/template-linter.ts#L44-L47):

```js
const errors = linter.verify({
  source,
  moduleId: textDocument.uri
});
```

However, [`bin/ember-template-lint.js` seems to pass `filePath` *and* `moduleId`](https://github.com/ember-template-lint/ember-template-lint/blob/706eac0674cc04a210171c21d95ebb628acc23e3/bin/ember-template-lint.js#L29).

Digging deeper, it also seems that `getConfigForFile` [only takes `overrides` into account if `filePath` is provided](https://github.com/ember-template-lint/ember-template-lint/blob/706eac0674cc04a210171c21d95ebb628acc23e3/lib/get-config.js#L416).

I also notice [this tell-tale TODO](https://github.com/ember-template-lint/ember-template-lint/blame/639db0d160b6a1878f325ed0c77f57a411ccaaed/lib/linter.js#L292):

```js
// TODO: deprecate moduleId and moduleName
```

Which is the correct option to pass? `moduleId`? `filePath`? Both?

**Comment by rwjblue** [comment]
tldr; `filePath` (when representing the _real on disk_ path) is the new hawtness, `moduleId` is the old stinker.

Certain features will not work when not using `filePath` (e.g. `overrides`), but since some rules may have been relying on the presence of `moduleId` the bin script still passes both `moduleId` and `filePath`. 

As mentioned in that TODO comment, we need to deprecate using `moduleId`. This can be done by:

* emitting a warning when the linter is only passed a `moduleId` and not a `filePath`
* emitting a warning when a rule accesses `moduleId` off its constructor args (likely via a getter with a deprecation)

Both of those should emit only one warning for the duration of the run (e.g. it shouldn't emit a console warning for every template / instantiation done).

**Comment by jgwhite** [comment]
Perfect, thanks @rwjblue!

**Comment by rwjblue** [comment]
Reopening to track getting things updated a bit (we at least need to update the `README.md` docs, even if we delay the actual deprecation).
